undefined reference to `_gfortran_set_options'

If I take our small `examples/expr2.f90`, but name if `expr.F90`, I get the following (linking) error:
```
$ lfortran expr2.F90
/usr/bin/ld: /tmp/expr2-1ed0b6.o: in function `MAIN__':
expr2.F90:(.text+0x46): undefined reference to `_gfortran_st_write'
/usr/bin/ld: expr2.F90:(.text+0x61): undefined reference to `_gfortran_transfer_integer_write'
/usr/bin/ld: expr2.F90:(.text+0x70): undefined reference to `_gfortran_st_write_done'
/usr/bin/ld: /tmp/expr2-1ed0b6.o: in function `main':
expr2.F90:(.text+0x93): undefined reference to `_gfortran_set_args'
/usr/bin/ld: expr2.F90:(.text+0xa2): undefined reference to `_gfortran_set_options'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
The command 'clang -o expr2.out expr2.F90  -L"src/bin/../runtime" -Wl,-rpath,"src/bin/../runtime" -llfortran_runtime -lm' failed.
```

Any idea?

--- a/.github/workflows/CI.yml
+++ b/.github/workflows/CI.yml
@@ -341,13 +341,13 @@ jobs:
         shell: bash -e -l {0}
         run: |
           mv inst inst.moved
-          cp src/examples/expr2.f90 .
+          cp src/examples/expr2.f90 expr2.F90
           rm -rf src
-          inst.moved/bin/lfortran expr2.f90 -o expr2
+          inst.moved/bin/lfortran expr2.F90 -o expr2
           ./expr2
-          inst.moved/bin/lfortran --backend=c expr2.f90 -o expr2b
+          inst.moved/bin/lfortran --backend=c expr2.F90 -o expr2b
           ./expr2b
-          inst.moved/bin/lfortran --openmp --backend=cpp expr2.f90 -o expr2c
+          inst.moved/bin/lfortran --openmp --backend=cpp expr2.F90 -o expr2c
           ./expr2c
 
   debug:

--- a/src/bin/lfortran.cpp
+++ b/src/bin/lfortran.cpp
@@ -2324,7 +2324,7 @@ int main_app(int argc, char *argv[]) {
         }
     }
 
-    if (endswith(arg_file, ".f90") || endswith(arg_file, ".f")) {
+    if (endswith(arg_file, ".f90") || endswith(arg_file, ".f") || endswith(arg_file, ".F90") || endswith(arg_file, ".F")) {
         if (backend == Backend::x86) {
             return compile_to_binary_x86(arg_file, outfile,
                     time_report, compiler_options);

